src/general/world_effects/PhotosynthesisProductionEffect.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Collections.Generic;
+using Godot;
+using Newtonsoft.Json;
+using Systems;
+
+/// <summary>
+///   Creates oxygen based on photosynthesizers (and removes carbon). And does the vice versa for oxygen consumption
+///   to balance things out. This is kind of a simplified version of <see cref="CompoundProductionEffect"/>
+/// </summary>
+[JSONDynamicTypeAllowed]
+public class PhotosynthesisProductionEffect : IWorldEffect
+{
+    [JsonProperty]
+    private GameWorld targetWorld;
+
+    public PhotosynthesisProductionEffect(GameWorld targetWorld)
+    {
+        this.targetWorld = targetWorld;
+    }
+
+    public void OnRegisterToWorld()
+    {
+    }
+
+    public void OnTimePassed(double elapsed, double totalTimePassed)
+    {
+        ApplyCompoundsAddition();
+    }
+
+    private void ApplyCompoundsAddition()
+    {
+        var outputModifier = 1.0f;
+        var inputModifier = 1.0f;
+        var modifier = 0.0001f;
+
+        List<TweakedProcess> microbeProcesses = [];
+
+        foreach (var patchKeyValue in targetWorld.Map.Patches)
+        {
+            var patch = patchKeyValue.Value;
+
+            // Skip empty patches as they don't need handling
+            if (patch.SpeciesInPatch.Count < 1)
+                continue;
+
+            float oxygenBalance = 0;
+            float co2Balance = 0;
+
+            foreach (var species in patch.SpeciesInPatch)
+            {
+                // Only microbial photosynthesis and respiration are taken into account
+                if (species.Key is not MicrobeSpecies microbeSpecies)
```

Comment:
There's already so many layers of indentation here that I would just split stuff to a separate method if another layer of indents was needed here. So that's why I use a guard clause here. You are of course right that this code structure won't work when multicellular stuff is added, but considering how long it has been with still 0 support in auto-evo for multicellular, I don't see needing to change this code for multiple years.